users/actions/activate_user.py

```diff
@@ -4,21 +4,20 @@
 from users.models import User
 
 
-class ActivateUser:
+class ActivateUser(object):
     def __init__(self, token_generator, uidb64, token):
         self.token_generator = token_generator
         self.token = token
-
         try:
             # urlsafe_base64_decode() decodes to bytestring on Python 3
             uid = force_text(urlsafe_base64_decode(uidb64))
             self.user = User.objects.get(pk=uid)
         except (TypeError, ValueError, OverflowError, User.DoesNotExist):
             self.user = None
 
-    def is_valid(self):
-        return self.user is not None and self.token_generator.check_token(self.user, self.token)
-
-    def call(self):
+    def __call__(self):
```

Comment:
So then python is not designed explicitely enough. That's why we have __call__ magic method, so we dont have to re-implement java.